### PR TITLE
Sonarlint91

### PR DESCRIPTION
**Pull request by sfuhrm**
Make sure that:

- [x] You have read the [contribution guidelines](https://github.com/mp911de/logstash-gelf/blob/master/.github/CONTRIBUTING.md).
- [x] You use the code formatters provided [here](https://github.com/mp911de/logstash-gelf/blob/master/as7formatter.xml) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.

 

This reduces the number of sonarlint issues in the main code to 91.

### CODE REVIEW

File: src/main/java/biz/paluch/logging/gelf/GelfMessageAssembler.java [hunk 1]
```diff
@@ -86,43 +86,43 @@ public void initialize(PropertyProvider propertyProvider) {
             host = propertyProvider.getProperty(PropertyProvider.PROPERTY_GRAYLOG_HOST);
         }
 
-        String port = propertyProvider.getProperty(PropertyProvider.PROPERTY_PORT);
-        if (port == null) {
-            port = propertyProvider.getProperty(PropertyProvider.PROPERTY_GRAYLOG_PORT);
+        String myPort = propertyProvider.getProperty(PropertyProvider.PROPERTY_PORT);
```
Comment (mp911de) [inline src/main/java/biz/paluch/logging/gelf/GelfMessageAssembler.java]: Introducing `my…` can help to solve field shadowing. At the same time, it contributes to reading complexity as that's not the style of our code.

File: src/main/java/biz/paluch/logging/gelf/Values.java [hunk 2]
```diff
@@ -13,35 +13,35 @@
  */
 public class Values {
 
-    private Map<String, Object> values = new HashMap<>();
+    private Map<String, Object> keyValueMap = new HashMap<>();
```
Comment (mp911de) [inline src/main/java/biz/paluch/logging/gelf/Values.java]: Not sure this is a helpful change.

File: src/main/java/biz/paluch/logging/gelf/intern/GelfMessage.java [hunk 3]
```diff
@@ -309,15 +309,15 @@ static Object getAdditionalFieldValue(String value, String fieldType) {
 
         if (messageBytes.length > maximumMessageSize) {
             // calculate the length of the datagrams array
-            int datagrams_length = messageBytes.length / maximumMessageSize;
+            int datagramsLength = messageBytes.length / maximumMessageSize;
```
Comment (mp911de) [inline src/main/java/biz/paluch/logging/gelf/intern/GelfMessage.java]: :+1:

File: src/main/java/biz/paluch/logging/gelf/intern/PoolHolder.java [hunk 4]
```diff
@@ -40,7 +40,7 @@ public static PoolHolder threadLocal() {
 
     private static class NoOpPoolHolder extends PoolHolder {
 
-        private final static NoOpPoolHolder instance = new NoOpPoolHolder();
+        private static final NoOpPoolHolder instance = new NoOpPoolHolder();
```
Comment (mp911de) [inline src/main/java/biz/paluch/logging/gelf/intern/PoolHolder.java]: :+1:

File: src/main/java/biz/paluch/logging/gelf/Values.java (hunk 2 above)
Comment (sfuhrm) [reply src/main/java/biz/paluch/logging/gelf/Values.java]: I know what you mean ... naming a field with the same name of the class was the SonarLint issue. I think no change is better than just making a tool happy.

File: src/main/java/biz/paluch/logging/gelf/GelfMessageAssembler.java (hunk 1 above)
Comment (sfuhrm) [reply src/main/java/biz/paluch/logging/gelf/GelfMessageAssembler.java]: Yes, this is for sure a question of taste IMHO. Shadowing doesn't hurt if you're a bit experienced and you have an IDE. 
I can live with shadowing and let SonarLint detect this.
Some projects insist on having all static code analysis metrics green :-).

**Comment by sfuhrm** [comment]
Reverted the unshadowing of fields and the map field name.

**Comment by mp911de** [comment]
Thanks a lot, that's merged now.

**Pull request merged**